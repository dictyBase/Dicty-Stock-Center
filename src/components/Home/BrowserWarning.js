// @flow
import React, { Component } from "react"
import Button from "material-ui/Button"
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle
} from "material-ui/Dialog"
import Slide from "material-ui/transitions/Slide"

type Props = {}

type State = {
  /** Tells if the dialog box is open or not */
  open: boolean
}

const Transition = props => {
  return <Slide direction="up" {...props} />
}

/**
 * This is a browser warning component that tells the user to upgrade if they are using an outdated version (i.e. IE10 or below). It uses Material-UI's Dialog component and Slide animation.
 */

export default class BrowserWarning extends Component<Props, State> {
  state = {
    open: true
  }
  handleClose = () => {
    this.setState({ open: false })
  }
  render() {
    return (
      <div>
        <Dialog
          open={this.state.open}
          transition={Transition}
          keepMounted
          onClose={this.handleClose}
          aria-labelledby="alert-dialog-slide-title"
          aria-describedby="alert-dialog-slide-description">
          <DialogTitle id="alert-dialog-slide-title">
            {"You are using an outdated browser!"}
          </DialogTitle>
          <DialogContent>
            <DialogContentText id="alert-dialog-slide-description">
              <p>
                Dicty Stock Center may not work as expected on your browser.
              </p>
              <p>
                Please consider upgrading to a more modern browser such as{" "}
                <a
                  href="https://www.mozilla.org/en-US/firefox/new/"
                  target="_blank"
                  rel="noopener noreferrer">
                  Mozilla Firefox
                </a>{" "}
                or{" "}
                <a
                  href="https://www.google.com/chrome/"
                  target="_blank"
                  rel="noopener noreferrer">
                  Google Chrome
                </a>.
              </p>
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClose} color="primary">
              Close
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    )
  }
}
