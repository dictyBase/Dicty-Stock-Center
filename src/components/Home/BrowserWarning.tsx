import React, { useState } from "react"
import Button from "@material-ui/core/Button"
import Dialog from "@material-ui/core/Dialog"
import DialogActions from "@material-ui/core/DialogActions"
import DialogTitle from "@material-ui/core/DialogTitle"
import Slide from "@material-ui/core/Slide"
import BrowserWarningContent from "./BrowserWarningContent"

const Transition = props => <Slide direction="up" {...props} />

/**
 * BrowserWarning tells the user to upgrade if they are using an outdated version (i.e. IE10 or below).
 * It uses Material-UI's Dialog component and Slide animation.
 */

const BrowserWarning = () => {
  const [open, setOpen] = useState(true)

  const handleClose = () => {
    setOpen(false)
  }

  return (
    <>
      <Dialog
        open={open}
        TransitionComponent={Transition}
        keepMounted
        onClose={handleClose}>
        <DialogTitle id="outdated-browser-warning">
          {"You are using an outdated browser!"}
        </DialogTitle>
        <BrowserWarningContent />
        <DialogActions>
          <Button onClick={handleClose} color="primary">
            Close
          </Button>
        </DialogActions>
      </Dialog>
    </>
  )
}

export default BrowserWarning
