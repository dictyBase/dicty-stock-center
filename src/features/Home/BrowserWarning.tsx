import React, { useState } from "react"
import Button from "@material-ui/core/Button"
import Dialog from "@material-ui/core/Dialog"
import DialogActions from "@material-ui/core/DialogActions"
import DialogTitle from "@material-ui/core/DialogTitle"
import Typography from "@material-ui/core/Typography"
import BrowserWarningContent from "./BrowserWarningContent"

/**
 * BrowserWarning tells the user to upgrade if they are using an outdated browser and/or version.
 */

const BrowserWarning = () => {
  const [open, setOpen] = useState(true)

  const handleClose = () => {
    setOpen(false)
  }

  return (
    <div>
      <Dialog open={open} keepMounted onClose={handleClose}>
        <DialogTitle disableTypography id="outdated-browser-warning">
          <Typography variant="h1">
            You are using an outdated browser!
          </Typography>
        </DialogTitle>
        <BrowserWarningContent />
        <DialogActions>
          <Button onClick={handleClose} color="primary">
            Close
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}

export default BrowserWarning
