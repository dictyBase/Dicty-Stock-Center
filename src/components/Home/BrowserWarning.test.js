import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import BrowserWarning from "./BrowserWarning"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<BrowserWarning />, div)
})

test("matching a snapshot of BrowserWarning", () => {
  const component = renderer.create(<BrowserWarning />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
