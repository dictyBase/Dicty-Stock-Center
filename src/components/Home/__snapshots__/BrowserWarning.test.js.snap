// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of BrowserWarning 1`] = `<div />`;
